pkg/networkserver/utils.go

```diff
@@ -501,7 +501,7 @@ func newMACState(dev *ttnpb.EndDevice, fps *frequencyplans.Store, defaults ttnpb
 	macState.CurrentParameters.ADRDataRateIndex = ttnpb.DATA_RATE_0
 	macState.DesiredParameters.ADRDataRateIndex = macState.CurrentParameters.ADRDataRateIndex
 
-	macState.CurrentParameters.ADRTxPowerIndex = 0
+	macState.CurrentParameters.ADRTxPowerIndex = uint32(phy.MaxTxPowerIndex)
```

Comment:
> Assume `MaxTxPowerIndex` is the default `TxPowerIndex` as per LoRaWAN spec.

Where does the spec say this? As far as I remember, "vanilla" end devices start at `TxPowerIndex` 0 (the `TxPowerIndex` that corresponds to the maximum `TxPower`).

Reply:
I was referring to this paragraph in 1.0.4 spec: [1.0.4](https://imgur.com/jJeN5oJ).
Maximum transmission power is `TxPowerIndex==0` however indeed.